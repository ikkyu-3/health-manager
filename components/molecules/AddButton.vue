<template>
  <v-btn fab dark :color="color" @click="click">
    <v-icon>fas fa-plus</v-icon>
  </v-btn>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'

@Component
export default class AddButton extends Vue {
  @Prop({ default: 'teal' }) readonly color!: string

  click(e: MouseEvent): void {
    this.$emit('click', e)
  }
}
</script>
