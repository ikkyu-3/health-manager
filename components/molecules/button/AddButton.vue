<template>
  <v-btn fab dark :color="color" :class="className" @click="click">
    <v-icon>fa-plus</v-icon>
  </v-btn>
</template>

<script lang="ts">
import { createComponent } from '@vue/composition-api'

export default createComponent({
  props: {
    color: {
      type: String,
      default: 'teal'
    }
  },
  setup(_, { attrs, emit }) {
    const className = attrs.hasOwnProperty('right-bottom') ? 'rightBottom' : ''

    function click(e: MouseEvent) {
      emit('click', e)
    }

    return { className, click }
  }
})
</script>

<style lang="scss" scoped>
.rightBottom {
  position: absolute;
  right: 16px;
  bottom: 16px;
}
</style>
