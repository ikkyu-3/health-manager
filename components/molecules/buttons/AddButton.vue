<template>
  <v-btn
    fab
    dark
    :color="color"
    :x-small="state.xSmall"
    :small="state.small"
    :large="state.large"
    :x-large="state.xLarge"
    @click="click"
  >
    <v-icon>fa-plus</v-icon>
  </v-btn>
</template>

<script lang="ts">
import { createComponent, reactive } from '@vue/composition-api'

export default createComponent({
  props: {
    color: {
      type: String,
      default: 'grey'
    }
  },
  setup(_, { attrs, emit }) {
    const state = reactive({
      xSmall: attrs.hasOwnProperty('x-small'),
      small: attrs.hasOwnProperty('small'),
      large: attrs.hasOwnProperty('large'),
      xLarge: attrs.hasOwnProperty('x-large')
    })

    function click(e: MouseEvent) {
      emit('click', e)
    }

    return { state, click }
  }
})
</script>
