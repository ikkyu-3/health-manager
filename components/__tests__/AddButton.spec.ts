import { mount } from '@vue/test-utils'
import AddButton from '@/components/molecules/AddButton.vue'

describe('Molecules/AddButton.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(AddButton, {
      propsData: {
        color: undefined
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
