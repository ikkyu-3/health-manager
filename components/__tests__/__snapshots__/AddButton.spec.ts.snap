// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/AddButton.vue snapshot 1`] = `"<button type=\\"button\\" class=\\"v-btn v-btn--floating theme--dark teal\\"><div class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon fas fa-plus theme--dark\\"></i></div></button>"`;
